### PR TITLE
Exercise spotVMOptions in E2E tests

### LINKED ISSUE #4454: e2e tests should exercise `spotVMOptions`
/kind bug

#4453 suggests that CAPZ isn't working right with spot instances currently. Once it merges, we should add an e2e test that populates `spotVMOptions` to ensure this functionality doesn't break again.

**What did you expect to happen:**

Spot instances work in CAPZ without incident.

**Anything else you would like to add:**

### PR DESCRIPTION
**Pull request by willie-yao**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
This PR adds an optional E2E test for `spotVMOptions` to verify it does not break.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4454

**Special notes for your reviewer**:

- [x] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: templates/test/ci/prow-spot/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  - ../../../addons/cluster-api-helm/cloud-provider-azure-ci.yaml
+patchesStrategicMerge:
+  - ../patches/tags.yaml
+  - ../patches/mhc.yaml
+  - ../patches/controller-manager.yaml
+  - patches/spot-vm-options.yaml
+  - ../patches/uami-md-0.yaml
+  - ../patches/uami-control-plane.yaml
+  - ../patches/cluster-label-calico.yaml
+  - ../patches/cluster-label-cloud-provider-azure.yaml
```
Comment (nawazkh) [inline templates/test/ci/prow-spot/kustomization.yaml:19]: nit: new line at the end of the file

**Review by nawazkh — COMMENTED** [review]
minor non blocking nit, otherwise looks good to me. Pending tests of course.
/lgtm

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Pull request merged**